Release 73.0.0 (#1630)
MetaMask/core: packages/transaction-controller/CHANGELOG.md

## [Unreleased]

## [9.0.0]
### Added

> - **BREAKING**: Add required `getSelectedAddress` callback argument to constructor ([#1579](https://github.com/MetaMask/core/pull/1579))

Reviewer: Nit: I can see how it makes sense to phrase this change entry as "Added required ___", but it feels more appropriate under the "Changed" categories. Added is more often used for "new features" that can be taken advantage of, whereas this is a change in expectations for the caller. Likewise for the second entry as well.

Author: Great point, done.

- Add `baseFeePerGas` to transaction metadata ([#1590](https://github.com/MetaMask/core/pull/1590))
- Add `txReceipt` to transaction metadata ([#1592](https://github.com/MetaMask/core/pull/1592))
- Add `initApprovals` method to generate approval requests from unapproved transactions ([#1575](https://github.com/MetaMask/core/pull/1575))
- Add optional `incomingTransactions` constructor arguments ([#1579](https://github.com/MetaMask/core/pull/1579))
  - `apiKey`
  - `includeTokenTransfers`
  - `isEnabled`
  - `updateTransactions`
- Add incoming transaction methods ([#1579](https://github.com/MetaMask/core/pull/1579))
  - `startIncomingTransactionPolling`
  - `stopIncomingTransactionPolling`
  - `updateIncomingTransactions` 
- Add `requireApproval` option to `addTransaction` method options ([#1580](https://github.com/MetaMask/core/pull/1580))
- Add `address` argument to `wipeTransactions` method ([#1573](https://github.com/MetaMask/core/pull/1573))

### Changed
- **BREAKING**: Move all but first argument to options bag in `addTransaction` method ([#1576](https://github.com/MetaMask/core/pull/1576))

> - **BREAKING**: Update `fromBlock` and `limit` types in `RemoteTransactionSourceRequest` ([#1579](https://github.com/MetaMask/core/pull/1579))

Reviewer: Nit: Reading this from the perspective of a mobile/extension maintainer, it's unclear how these have changed, or what actions I should take to accommodate these changes. If the "recommended action" for a breaking change isn't obvious, additional detail is greatly appreciated.

In this case it seems kinda moot, because it's a type that seems primarily meant for internal use. But from reading the changelog, it seemed like a change I'd need to handle at first.

Here's an attempt to explain that better, and also it corrects a mistake regarding the `limit` property (`limit` was unaffected, it was `networkType` that was removed)

```suggestion
- **BREAKING**: Update properties of `RemoteTransactionSourceRequest` type ([#1579](https://github.com/MetaMask/core/pull/1579))
  - The `fromBlock` property has changed from `string` to `number`
  - The `networkType` property has been removed
  - This type is intended mainly for internal use, so it's likely this change doesn't affect most projects
```

Author: I wasn't sure how verbose we should be in these changelogs, but this is indeed much clearer, done.

### Removed

> - **BREAKING**: Remove `fetchAll` method ([#1579](https://github.com/MetaMask/core/pull/1579))

Reviewer: Here's an attempt to explain in more detail how this breaking change should be handled:
```suggestion
- **BREAKING**: Remove `fetchAll` method ([#1579](https://github.com/MetaMask/core/pull/1579))
  - This method was used to fetch transaction history from Etherscan. This is now handled automatically by the controller instead; new transactions are fetched for each new block if polling is enabled.
  - Polling can be enabled or disabled by calling `startIncomingTransactionPolling` or `stopIncomingTransactionPolling` respectively.
  - An immediate update can be requested by calling `updateIncomingTransactions`
```

Author: Beautiful, done.

> - **BREAKING**: Remove `prepareUnsignedEthTx` and `getCommonConfiguration` methods ([#1581](https://github.com/MetaMask/core/pull/1581))

Reviewer: Additional detail about the change:
```suggestion
- **BREAKING**: Remove `prepareUnsignedEthTx` and `getCommonConfiguration` methods ([#1581](https://github.com/MetaMask/core/pull/1581))
  - - These methods were intended mainly for internal use, so it's likely this change doesn't affect most projects
```

Author: Done.

I've left the additional line as it does indeed add some noise but I think the relief and info for the future adopter is by far the lesser of two evils.

(We're also adopting this in the clients but I think it's a great standard to follow)

Reviewer: Yeah I was on the fence about it as well! But this seems OK for now.

It felt wrong to leave public changes undocumented, even if we weren't aware of any usage of them. But it also felt misleading to list them here with no context, as reviewers might not realize that it doesn't impact them.
